[SVG] Fix computed value for `marker` element

Ahmad-S792 opened the pull request:
#### 71b28fef35b71f617dbe5051ae54139c91067da0
<pre>
[SVG] Fix computed value for `marker` element
<a href="https://bugs.webkit.org/show_bug.cgi?id=294206">https://bugs.webkit.org/show_bug.cgi?id=294206</a>
<a href="https://rdar.apple.com/153373245">rdar://153373245</a>

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

In this patch, we do `shorthand` value as &apos;Single&apos; for &apos;marker&apos; itself,
which fixes computed value which was previously being skipped. As drive-by,
we also make it &apos;svg&apos; tagged.

Specification: <a href="https://svgwg.org/svg2-draft/painting.html#MarkerShorthand">https://svgwg.org/svg2-draft/painting.html#MarkerShorthand</a>

* LayoutTests/imported/w3c/web-platform-tests/svg/painting/parsing/marker-computed-expected.txt: Progression
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks-expected.txt:
* Source/WebCore/css/CSSProperties.json:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/71b28fef35b71f617dbe5051ae54139c91067da0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107882 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27550 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17968 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113096 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58407 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109848 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28245 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36096 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81882 "Found 4 new test failures: imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks.html imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/margin-collapsing-quirks/multicol-quirks-mode.html imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/margin-collapsing-quirks/multicol-standards-mode.html imported/w3c/web-platform-tests/svg/painting/parsing/marker-computed.svg (failure)") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110830 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22375 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97196 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62311 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21790 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15329 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57849 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91726 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15395 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116220 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34954 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25712 "Found 33 new test failures: fast/forms/color/input-color-swatch-overlay-appearance.html imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks.html imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/margin-collapsing-quirks/multicol-quirks-mode.html imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/margin-collapsing-quirks/multicol-standards-mode.html imported/w3c/web-platform-tests/svg/painting/parsing/marker-computed.svg inspector/css/add-css-property.html inspector/css/css-property.html inspector/css/forcePseudoState.html inspector/css/generateFormattedText.html inspector/css/getComputedPrimaryFontForNode.html ... (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90913 "Found 32 new test failures: imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks.html imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/margin-collapsing-quirks/multicol-quirks-mode.html imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/margin-collapsing-quirks/multicol-standards-mode.html imported/w3c/web-platform-tests/svg/painting/parsing/marker-computed.svg inspector/css/add-css-property.html inspector/css/css-property.html inspector/css/forcePseudoState.html inspector/css/generateFormattedText.html inspector/css/getComputedStyleForNode.html inspector/css/getMatchedStylesForNodeBackdropPseudoId.html ... (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35330 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93474 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90706 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35609 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13363 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30707 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34855 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40407 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34597 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37958 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36259 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d2dc7c52a06944bc0896d39c73949632fb5f5f70. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46758

Ahmad-S792 committed:
[SVG] Fix computed value for `marker` element
https://bugs.webkit.org/show_bug.cgi?id=294206
rdar://153373245

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

In this patch, we do `shorthand` value as 'Single' for 'marker' itself,
which fixes computed value which was previously being skipped. As drive-by,
we also make it 'svg' tagged.

Specification: https://svgwg.org/svg2-draft/painting.html#MarkerShorthand

* LayoutTests/imported/w3c/web-platform-tests/svg/painting/parsing/marker-computed-expected.txt: Progression
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/cssom-getPropertyValue-common-checks-expected.txt:
* Source/WebCore/css/CSSProperties.json:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/71b28fef35b71f617dbe5051ae54139c91067da0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46758

Ahmad-S792 commented:
Bunch of assertion failures:

```
ASSERTION FAILED: shorthand.length() == 1
/Volumes/Data/worker/macOS-Sequoia-Debug-Build-EWS/build/Source/WebCore/style/StyleExtractorCustom.h(838) : void WebCore::Style::extractSingleShorthandSerialization(ExtractorState &, StringBuilder &, const CSS::SerializationContext &, const StylePropertyShorthand &)
1   0x302c8fc28 WebCore::Style::extractSingleShorthandSerialization(WebCore::Style::ExtractorState&, WTF::StringBuilder&, WebCore::CSS::SerializationContext const&, WebCore::StylePropertyShorthand const&)
2   0x302aecd58 WebCore::Style::ExtractorFunctions::extractMarkerShorthandSerialization(WebCore::Style::ExtractorState&, WTF::StringBuilder&, WebCore::CSS::SerializationContext const&)
3   0x302add90c WebCore::Style::ExtractorGenerated::extractValueSerialization(WebCore::Style::ExtractorState&, WTF::StringBuilder&, WebCore::CSS::SerializationContext const&, WebCore::CSSPropertyID)
4   0x3071b6800 WebCore::Style::Extractor::propertyValueSerializationInStyle(WebCore::RenderStyle const&, WebCore::CSSPropertyID, WebCore::CSS::SerializationContext const&, WebCore::CSSValuePool&, WebCore::RenderElement const*, WebCore::Style::ExtractorState::PropertyValueType) const
5   0x3071b662c WebCore::Style::Extractor::propertyValueSerialization(WebCore::CSSPropertyID, WebCore::CSS::SerializationContext const&, WebCore::Style::Extractor::UpdateLayout, WebCore::Style::ExtractorState::PropertyValueType) const
6   0x30433b810 WebCore::CSSComputedStyleDeclaration::getPropertyValue(WTF::String const&)
7   0x30570bde0 WebCore::InspectorStyle::collectProperties(bool)
8   0x30570b75c WebCore::InspectorStyle::buildArrayForComputedStyle()
9   0x3057666bc WebCore::InspectorCSSAgent::getComputedStyleForNode(int)
10  0x12ef12a60 Inspector::CSSBackendDispatcher::getComputedStyleForNode(long, WTF::RefPtr<WTF::JSONImpl::Object, WTF::RawPtrTraits<WTF::JSONImpl::Object>, WTF::DefaultRefDerefTraits<WTF::JSONImpl::Object>>&&)
11  0x12ef11ba8 Inspector::CSSBackendDispatcher::dispatch(long, WTF::String const&, WTF::Ref<WTF::JSONImpl::Object, WTF::RawPtrTraits<WTF::JSONImpl::Object>, WTF::DefaultRefDerefTraits<WTF::JSONImpl::Object>>&&)
12  0x12ef0a54c Inspector::BackendDispatcher::dispatch(WTF::String const&)
13  0x3056c2734 WebCore::InspectorController::dispatchMessageFromFrontend(WTF::String const&)
14  0x3056dcbe8 WebCore::InspectorBackendDispatchTask::dispatchOneMessage()
15  0x3056dcae4 WebCore::InspectorBackendDispatchTask::scheduleOneShot()::'lambda'()::operator()() const
16  0x3056dc848 WTF::Detail::CallableWrapper<WebCore::InspectorBackendDispatchTask::scheduleOneShot()::'lambda'(), void>::call()
17  0x12d604dcc WTF::Function<void ()>::operator()() const
18  0x12d6d10dc WTF::RunLoop::performWork()
19  0x12d6d5320 WTF::RunLoop::performWork(void*)
20  0x1834b9cd4 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
21  0x1834b9c68 __CFRunLoopDoSource0
22  0x1834b99d4 __CFRunLoopDoSources0
23  0x1834b8628 __CFRunLoopRun
24  0x1834b7c58 CFRunLoopRunSpecific
25  0x184a86c78 -[NSRunLoop(NSRunLoop) runMode:beforeDate:]
26  0x184afe79c -[NSRunLoop(NSRunLoop) run]
27  0x1830e09c8 _xpc_objc_main
28  0x1830f0fc8 _xpc_main
29  0x1830e0568 _xpc_copy_xpcservice_dictionary
30  0x1193cb00c WebKit::XPCServiceMain(int, char const**)
31  0x11c29f730 WKXPCServiceMain
com.apple.WebKit.WebContent.Development terminated (pid 11527) for reason: crash
LEAK: 2 WebPageProxy
```

WebKit Source: https://github.com/WebKit/WebKit/blob/1eaa89c16e4430f88eaec63a5427d69f0d2bd593/Source/WebCore/style/StyleExtractorCustom.h#L838

@weinig - Any idea, if I am using this new thing correctly or doing something wrong? First time using auto-generated part, so would be good to get some insight and help.
